lib/avro_ex/encode.ex

```diff
@@ -157,12 +157,15 @@ defmodule AvroEx.Encode do
       size ->
         acc = do_encode(%Primitive{type: :long}, context, size)
 
-        Enum.reduce(map, acc, fn {k, v}, acc ->
-          key = do_encode(%Primitive{type: :string}, context, k)
-          value = do_encode(values, context, v)
+        encoded_map =
+          Enum.reduce(map, acc, fn {k, v}, acc ->
+            key = do_encode(%Primitive{type: :string}, context, k)
+            value = do_encode(values, context, v)
 
-          acc <> key <> value
-        end)
+            acc <> key <> value
+          end)
+
+        encoded_map <> <<0>>
```

Comment:
The spec doesn't mention a 0 terminator for [maps](https://avro.apache.org/docs/current/spec.html#map_encoding) like it does with [arrays](https://avro.apache.org/docs/current/spec.html#array_encoding). Have you tested these changes to the map implementation yet?

Reply:
Update: Ignore the question about testing maps. You tested [here](https://github.com/rentpath/avro_ex/pull/2/files#diff-c76d23b06408448815c816c656eff2ccR281). I'm wondering why the zero is on the end, though. Do you know?

Reply:
> Maps are encoded as a series of blocks. Each block consists of a long count value, followed by that many key/value pairs. A block with count zero indicates the end of the map. Each item is encoded per the map's value schema.

Reply:
💡 I hadn't understood what that part of the spec meant. I think it makes more sense now.